data/model/namespacequota.py

```diff
@@ -0,0 +1,425 @@
+import json
+
+from peewee import fn, JOIN
+
+from data.database import (
+    Repository,
+    QuotaLimits,
+    UserOrganizationQuota,
+    QuotaType,
+    Manifest,
+    get_epoch_timestamp_ms,
+    Tag,
+    RepositorySize,
+    User,
+)
+from data import model
+from data.model import (
+    organization,
+    user,
+    InvalidUsernameException,
+    notification,
+    config,
+    InvalidSystemQuotaConfig,
+)
+
+
+def verify_namespace_quota(repository_ref):
+    model.repository.get_repository_size_and_cache(repository_ref._db_id)
+    namespace_size = get_namespace_size(repository_ref.namespace_name)
+    return check_limits(repository_ref.namespace_name, namespace_size)
+
+
+def verify_namespace_quota_force_cache(repository_ref):
+    force_cache_repo_size(repository_ref)
+    namespace_size = get_namespace_size(repository_ref.namespace_name)
+    return check_limits(repository_ref.namespace_name, namespace_size)
+
+
+def verify_namespace_quota_during_upload(repository_ref):
+    size = model.repository.get_size_during_upload(repository_ref._db_id)
+    namespace_size = get_namespace_size(repository_ref.namespace_name)
+    if namespace_size is None:
+        namespace_size = 0
+
+    return check_limits(repository_ref.namespace_name, size + namespace_size)
+
+
+def check_limits(namespace_name, size):
+    limits = get_namespace_limits(namespace_name)
+    limit_bytes = 0
+    severity_level = None
+    for limit in limits:
+        if size > limit["bytes_allowed"]:
+            if limit_bytes < limit["bytes_allowed"]:
+                limit_bytes = limit["bytes_allowed"]
+                severity_level = limit["type_id"]
+
+    return {"limit_bytes": limit_bytes, "severity_level": severity_level}
+
+
+def notify_organization_admins(repository_ref, notification_kind, metadata={}):
+    if repository_ref.namespace_user.organization:
+        org = organization.get_organization(repository_ref.namespace_user)
+        admins = organization.get_admin_users(org)
+
+        for admin in admins:
+            notification.create_notification(
+                notification_kind, admin, {"namespace": repository_ref.namespace_user}
+            )
+    else:
+        notification.create_notification(
+            notification_kind,
+            repository_ref.namespace_user,
+            {"namespace": repository_ref.namespace_user},
+        )
+
+
+def force_cache_repo_size(repository_ref):
+    return model.repository.force_cache_repo_size(repository_ref._db_id)
+
+
+def create_namespace_quota(name, limit_bytes):
+    user_object = user.get_namespace_user(name)
+
+    UserOrganizationQuota.create(namespace_id=user_object.id, limit_bytes=limit_bytes)
+
+    return UserOrganizationQuota
```

Comment:
You're returning the table class here. Instead of the created instance. Also need to check whether the org quota already exists or not, and handle it (do nothing, replace the existing one with the new value, ...?)

Reply:
This is true. The value returned is never used which is why this still functions. I will fix it.